addons/phantom_camera/scripts/phantom_camera_host/phantom_camera_host.gd

```diff
@@ -324,21 +324,34 @@ func _process(delta):
 #region Public Functions
 
 func _show_viewfinder_in_play() -> void:
-	if _active_pcam.show_viewfinder_in_play:
-		if not Engine.is_editor_hint() && OS.has_feature("editor"): # Only appears when running in the editor
-			var canvas_layer: CanvasLayer = CanvasLayer.new()
-			get_tree().get_root().get_child(0).add_child(canvas_layer)
-
-			if not is_instance_valid(_viewfinder_node):
-				var _viewfinder_scene := load("res://addons/phantom_camera/panel/viewfinder/viewfinder_panel.tscn")
-				_viewfinder_node = _viewfinder_scene.instantiate()
-				canvas_layer.add_child(_viewfinder_node)
-			else:
-				_viewfinder_node.visible = true
-				_viewfinder_node.update_dead_zone()
-	else:
-		if is_instance_valid(_viewfinder_node):
-			_viewfinder_node.visible = false
+	# Don't show the viewfinder in the actual editor or project builds
+	if Engine.is_editor_hint() or !OS.has_feature("editor"):
+		return
+
+	# We default the viewfinder node to hidden
+	if is_instance_valid(_viewfinder_node):
+		_viewfinder_node.visible = false
+
+	if !_active_pcam.show_viewfinder_in_play:
+		return
+
+	var _typed_active_pcam = (_active_pcam as PhantomCamera2D) if _is_2D else (_active_pcam as PhantomCamera3D)
+	assert(_typed_active_pcam != null, "The current active PCam couldn't be properly type-cast")
```

Comment:
Don't think this is needed?
There is a flag that validates whether if the `_active_pcam` is an instance of `PCam2D` / `PCam3D` before calling `_show_viewfinder_in_play()`.

Reply:
You're right. Since `_active_pcam` is of type `Node`, my LSP wasn't showing `PCam.follow_mode` and `PCam.FollowMode` as valid properties, but Godot has no problem running this even without the type cast. Will change this.

Oh, dynamic typing 😖.

Reply:
A blessing and a curse — all in one convenient language :smile: